build.rs

```diff
@@ -7,46 +7,24 @@ use std::path::Path;
 use std::process::Command;
 
 fn make_and_install(source_dir: &str, build_dir: &str, install_dir: &str) -> std::io::Result<()> {
-    let cmake_install_prefix = format!("-DCMAKE_INSTALL_PREFIX={}", install_dir);
-
-    for iceoryx_component in &["iceoryx_hoofs", "iceoryx_posh"] {
-        let component_source_dir = format!("{}/{}", source_dir, iceoryx_component);
-        let component_build_dir = format!("{}/{}", build_dir, iceoryx_component);
-
-        Command::new("mkdir")
-            .args(&["-p", &component_build_dir])
-            .output()
-            .map_err(|out| {
-                println!("{:?}", out);
-                out
-            })
-            .map(|out| println!("{:?}", out))?;
-
-        Command::new("cmake")
-            .current_dir(&component_build_dir)
-            .args(&[
-                "-DCMAKE_BUILD_TYPE=Release",
-                "-DBUILD_SHARED_LIBS=OFF",
-                &cmake_install_prefix,
-                &component_source_dir,
-            ])
-            .output()
-            .map_err(|out| {
-                println!("{:?}", out);
-                out
-            })
-            .map(|out| println!("{:?}", out))?;
-
-        Command::new("cmake")
-            .current_dir(&component_build_dir)
-            .args(&["--build", ".", "--target", "install"])
-            .output()
-            .map_err(|out| {
-                println!("{:?}", out);
-                out
-            })
-            .map(|out| println!("{:?}", out))?;
-    }
+    Command::new("cmake")
+        .current_dir(source_dir)
+        .arg(format!("-B{}", build_dir))
+        .arg("-Hiceoryx_meta")
+        .arg("-DCMAKE_BUILD_TYPE=Release")
+        .arg("-DBUILD_SHARED_LIBS=OFF")
```

Comment:
I'm not sure about this change. It adds some overhead and it should be possible to build the libs separately like we do it in ROS.

I can reproduce the error from #17 when I remove libacl from my system, even with your changes. So the problem you have might be something else. 

I also noticed that I used `std::process::Command` in the wrong way since I did not check for the status.

Could you use this code and share the log of a failed build
```rs
use std::io::{Error, ErrorKind};
fn make_and_install(source_dir: &str, build_dir: &str, install_dir: &str) -> std::io::Result<()> {
    let cmake_install_prefix = format!("-DCMAKE_INSTALL_PREFIX={}", install_dir);

    for iceoryx_component in &["iceoryx_hoofs", "iceoryx_posh"] {
        let component_source_dir = format!("{}/{}", source_dir, iceoryx_component);
        let component_build_dir = format!("{}/{}", build_dir, iceoryx_component);

        if !Command::new("mkdir")
            .args(&["-p", &component_build_dir])
            .status()?
            .success()
        {
            return Err(Error::new(
                ErrorKind::Other,
                format!("Could not create build dir for '{}'!", iceoryx_component),
            ));
        }

        if !Command::new("cmake")
            .current_dir(&component_build_dir)
            .args(&[
                "-DCMAKE_BUILD_TYPE=Release",
                "-DBUILD_SHARED_LIBS=OFF",
                &cmake_install_prefix,
                &component_source_dir,
            ])
            .status()?
            .success()
        {
            return Err(Error::new(
                ErrorKind::Other,
                format!("Could not run cmake for '{}'!", iceoryx_component),
            ));
        }

        if !Command::new("cmake")
            .current_dir(&component_build_dir)
            .args(&["--build", ".", "--target", "install"])
            .status()?
            .success()
        {
            return Err(Error::new(
                ErrorKind::Other,
                format!("Could not build '{}'!", iceoryx_component),
            ));
        }
    }

    Ok(())
}
```

Reply:
Please see the `build.log` attached to #21